src/coreclr/vm/frames.cpp

```diff
@@ -37,6 +37,255 @@
 
 #define CHECK_APP_DOMAIN    0
 
+void Frame::GcScanRoots(promote_func *fn, ScanContext* sc)
+{
+    switch (this->GetFrameIdentifier())
+    {
+#define FRAME_TYPE_NAME(frameType) case FrameIdentifier::frameType: { return dac_cast<PTR_##frameType>(this)->GcScanRoots_Impl(fn, sc); }
+#include "FrameTypes.h"
+    default:
+        UNREACHABLE();
```

Comment:
```suggestion
        DoJITFailFast();
```
UNREACHABLE tells the compiler that there is zero chance for this path to be taken and executing this path is UB.

This saves a few instructions (https://godbolt.org/z/j4qPz5K5b), but it exposes us to arbitrary code execution in case the frame identifier is corrupted. Given that we have deleted the GS cookie, we may want to keep the behavior defined for corrupted frame identifier.